### PR TITLE
[TIR][Schedule] Improve blockize to support blockizing multiple blocks

### PR DESCRIPTION
**Pull request by multiverstack-intellif**
This is to improve blockize primitive to support blockizing multiple blocks into a nested block. This helps when we want to schedule a big part of IR as a whole block.
Some possible simplification is performed during the blockize process, to keep TIR not that complex, so some other test case may need adjustment.

cc @Hzfengsy @junrushao @wrongtest-intellif

### CODE REVIEW

File: tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py [hunk 1]
```diff
@@ -988,15 +988,15 @@ def conv2d_1x1_0(inputs: T.Buffer((1, 16, 16, 64), "float16"), weight: T.Buffer(
                                 v2, v3 = T.axis.remap("SS", [ax2_1, ax3])
                                 v4_o = T.axis.spatial(1, 0)
                                 v5_o = T.axis.spatial(1, 0)
-                                T.reads(conv2d_nhwc_reindex_shared_wmma_accumulator[v0, v1, v2, v3, 0:16, 0:16])
-                                T.writes(conv2d_nhwc_reindex_shared[v0, v1, v2, v3, 0:16, 0:16])
+                                T.reads(conv2d_nhwc_reindex_shared_wmma_accumulator[v0, v1, 0, 0, 0:16, 0:16])
```
Comment (Hzfengsy) [inline tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py:991]: I think this kind of chengs may break current tensorcore support IMO. cc @junrushao @vinx13

Comment (vinx13) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py:991]: yes these unit iters can be important in some workloads (1x1 conv, or batch size = 1 gemm)

Comment (multiverstack-intellif) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py:991]: Thanks for the reminder. Do you mean tensorcore needs read/write ranges to be in a specific form? What kind of requirement does it like?

Comment (vinx13) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py:991]: it doesn't require specific forms of reads/writes. however, it needs iter vars with extent = 1 to be kept for dimensions used for padding

Comment (multiverstack-intellif) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py:991]: Thanks. Do you mean preserving the for loop with unit extend that `preserve_unit_iters` ensure? The for loop in this case is not changed, only the read/write range is changed here. Does it require all the iter vars be kept anywhere in reads/writes and expressions even if it's always constant 0?

File: python/tvm/tir/schedule/schedule.py [hunk 2]
```diff
@@ -2691,13 +2691,15 @@ def after_set_dtype(
     ########## Schedule: Blockize & Tensorize ##########
 
     @type_checked
-    def blockize(self, loop: LoopRV, preserve_unit_iters: bool = True) -> BlockRV:
+    def blockize(
+        self, target: Union[LoopRV, List[BlockRV]], preserve_unit_iters: bool = True
+    ) -> BlockRV:
         """Convert the subtree rooted at a specific loop into a block.
```
Comment (vinx13) [inline python/tvm/tir/schedule/schedule.py:2697]: need to update doc here

File: src/tir/schedule/primitive/blockize_tensorize.cc [hunk 3]
```diff
@@ -152,7 +152,8 @@ Array<Array<arith::IterMark>> SubspaceDivide(const BlockRealize& realize,
                                              const StmtSRef& block_sref,  //
```
Comment (vinx13) [inline src/tir/schedule/primitive/blockize_tensorize.cc:152]: update function doc for the new param

File: tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py (hunk 1 above)
Comment (vinx13) [reply tests/python/unittest/test_meta_schedule_schedule_rule_mlt_tc.py:991]: I verified these workloads and confirmed this PR doesn't break auto tensorization. Keeping these unit iters preserve more semantics of original IR, but I don't have strong opinion

File: src/tir/schedule/primitive/blockize_tensorize.cc (hunk 3 above)
Comment (multiverstack-intellif) [reply src/tir/schedule/primitive/blockize_tensorize.cc:157]: Thanks, updated now.

File: python/tvm/tir/schedule/schedule.py (hunk 2 above)
Comment (multiverstack-intellif) [reply python/tvm/tir/schedule/schedule.py]: Thanks, updated now.

File: src/tir/schedule/primitive/blockize_tensorize.cc [hunk 4]
```diff
@@ -219,30 +227,43 @@ Map<Var, PrimExpr> DeriveBlockBinding(const Array<IterVar>& iter_vars,
...
-    // substitution
     PrimExpr sub{nullptr};
-    if (is_one(outer_mark->extent)) {
-      sub = inner_iter->var;
+    if (is_one(inner_mark->extent)) {
+      // Skip inner var when extent is 1
+      // substitution
+      if (is_one(outer_mark->extent) && !preserve_unit_iters) {
+        // Simplify outer if not preserve_unit_iters
+        sub = outer_mark->extent - 1;
```
Comment (vinx13) [inline src/tir/schedule/primitive/blockize_tensorize.cc:250]: nit: is this equivalent to `make_zero(outer_mark.dtype()`? I feel this is more readable

**Review by vinx13 — APPROVED** [review]

File: src/tir/schedule/primitive/blockize_tensorize.cc (hunk 4 above)
Comment (multiverstack-intellif) [reply src/tir/schedule/primitive/blockize_tensorize.cc]: Yes, it's more direct, thanks.

**Pull request merged**